server/common_test.go

```diff
@@ -70,22 +106,34 @@ func runGin(s *server.Server) (*gin.Engine, error) {
 	return r, nil
 }
 
-type bblfshServiceMock struct {
-	ParseFunc              func(*protocol.ParseRequest) *protocol.ParseResponse
-	NativeParseFunc        func(*protocol.NativeParseRequest) *protocol.NativeParseResponse
-	VersionFunc            func(*protocol.VersionRequest) *protocol.VersionResponse
-	SupportedLanguagesFunc func(*protocol.SupportedLanguagesRequest) *protocol.SupportedLanguagesResponse
+type parseMock struct {
+	ParseFunc func(*protocol.ParseRequest) (*protocol.ParseResponse, error)
 }
 
-func (c *bblfshServiceMock) Parse(req *protocol.ParseRequest) *protocol.ParseResponse {
+func (c *parseMock) Parse(ctx xcontext.Context, req *protocol.ParseRequest) (resp *protocol.ParseResponse, gerr error) {
```

Comment:
is the named return needed here?

Reply:
copy past from service definition. Fixed.